packages/utilities/fast-web-utilities/src/array.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Returns the index of the last element in the array where predicate is true, and -1 otherwise.
+ *
+ * @param array - the array to test
+ * @param predicate - find calls predicate once for each element of the array, in descending order, until it finds one where predicate returns true. If such an element is found, findLastIndex immediately returns that element index. Otherwise, findIndex returns -1.
+ */
+export function findLastIndex<T>(
+    array: Array<T>,
+    predicate: (value: T, index: number, obj: T[]) => unknown
+): number {
+    if (!(predicate instanceof Function)) {
+        throw new TypeError();
+    }
+
+    let k = array.length;
+    while (k--) {
+        if (predicate(array[k], k, array)) {
+            return k;
+        }
+    }
+
+    return -1;
```

Comment:
You'll get better performance using [].reduce.

```suggestion
    return array.reduce((found, value, index, items) => predicate(value, index, items) ? index : found, -1);
```

Reply:
A `while` loop is much faster than a `reduce` here since the `return` will short-circuit as soon as a match is found.